From 496297fa1be912504d5091af8093f23abf9d77d2 Mon Sep 17 00:00:00 2001
From: Muyao Xu <muyaoxu@google.com>
Date: Wed, 13 Mar 2024 19:44:49 +0000
Subject: [PATCH] Fix CHECK() failure for fullscreen tabs started from another window

(cherry picked from commit 0fc99662b6f4e92dc13626b5b637fe24d01dbc9a)

Bug: 324562933
Change-Id: I240ca7be562d31528aee22d7a2df6a3a1c06f145
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5284115
Auto-Submit: Muyao Xu <muyaoxu@google.com>
Reviewed-by: Mike Wasserman <msw@chromium.org>
Commit-Queue: Muyao Xu <muyaoxu@google.com>
Cr-Original-Commit-Position: refs/heads/main@{#1259620}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5362422
Cr-Commit-Position: refs/branch-heads/6261@{#1069}
Cr-Branched-From: 9755d9d81e4a8cb5b4f76b23b761457479dbb06b-refs/heads/main@{#1250580}
---

--- a/chrome/browser/ui/exclusive_access/fullscreen_controller.cc
+++ b/chrome/browser/ui/exclusive_access/fullscreen_controller.cc
@@ -551,8 +551,8 @@
   if (chrome::IsRunningInAppMode())
     return;
 
-  CHECK(fullscreen_start_time_);
-  if (exclusive_access_tab()) {
+  // `fullscreen_start_time_` is null if a fullscreen tab moves to a new window.
+  if (fullscreen_start_time_ && exclusive_access_tab()) {
     ukm::SourceId source_id =
         exclusive_access_tab()->GetPrimaryMainFrame()->GetPageUkmSourceId();
     ukm::builders::Fullscreen_Exit(source_id)
